mender-update/cli/actions.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-update/cli/actions.hpp>
+
+namespace mender {
+namespace update {
+namespace cli {
+
+namespace conf = mender::common::conf;
+namespace database = mender::common::key_value_database;
+
+error::Error ShowArtifact(context::MenderContext &main_context) {
+	auto exp_provides = main_context.LoadProvides();
+	if (!exp_provides) {
+		return exp_provides.error();
+	}
+
+	auto &provides = exp_provides.value();
+	if (provides.count("artifact_name") == 0 || provides["artifact_name"] == "") {
```

Comment:
Q: Better safe than sorry, but at this point this is pretty much not a possibility, right?

Reply:
Well, the database can be corrupt in some way. You're right that it's not possible as long as the client is functioning correctly, but it's probably wise to treat it as untrusted in this context.